### PR TITLE
UI changes for Tarena students

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
Add UI changes for users with `features.Tarena = true`.

Includes refactor for logout.

## Key Changes

- Refactor logout in core/auth to use `User.logout`.
- Add `isTarena` feature flag check on User.
- Use the Tarena logo instead of CodeCombat
- Hide the top right nav bar for Tarena users.
- Hide the footer for Tarena users.

### CODE REVIEW

File: app/core/auth.coffee [hunk 1]
```diff
@@ -19,18 +19,11 @@ init = ->
 
   Backbone.listenTo me, 'sync', -> Backbone.Mediator.publish('auth:me-synced', me: me)
 
-module.exports.logoutUser = ->
+module.exports.logoutUser = (options={}) ->
   # TODO: Refactor to use User.logout
```
Comment (shubhi1092) [inline app/core/auth.coffee]: Can remove this comment

File: app/templates/base-flat.jade [hunk 2]
```diff
@@ -31,35 +31,44 @@ mixin accountLinks
...
-              a.navbar-brand(href="/home")
-                if serverConfig.codeNinjas
-                  img#logo-img.code-ninjas-powered-by(src="/images/pages/base/logo.png")
-                  img.code-ninjas-logo(src="/images/pages/base/code-ninjas-logo-right.png")
-                else
-                  img#logo-img(src="/images/pages/base/logo.png")
-                span.glyphicon.glyphicon-home
+              if me.useTarenaLogo()
+                a.navbar-brand(href="http://kidtts.tmooc.cn/ttsPage/login.html")
+                  img#logo-img(src="/images/pages/base/logo-tarena.png")
```
Comment (shubhi1092) [inline app/templates/base-flat.jade]: From asana, it seems we need to show codecombat logo as well as tarena logo next to it, similar to codeninjas (except that codecombat logo should not link to /home)

File: app/views/account/AccountSettingsView.coffee [hunk 3]
```diff
@@ -131,7 +131,7 @@ module.exports = class AccountSettingsView extends CocoView
           window?.webkit?.messageHandlers?.notification?.postMessage(name: "signOut") if window.application.isIPadApp
           Backbone.Mediator.publish("auth:logging-out", {})
           window.tracker?.trackEvent 'Log Out', category:'Homepage' if @id is 'home-view'
-          logoutUser($('#login-email').val())
+          logoutUser()
```
Comment (shubhi1092) [inline app/views/account/AccountSettingsView.coffee]: Why this change?

Comment (AndrewJakubowicz) [reply app/views/account/AccountSettingsView.coffee]: This argument was never used in the legacy code path and looks safe to remove.
I feel safer removing it as it was last changed 5 years ago.

**Comment by AndrewJakubowicz** [comment]
Thank you @shubhi1092 !
I'll fix the logo and I've answered your questions.

File: app/models/User.coffee [hunk 4]
```diff
@@ -508,6 +508,10 @@ module.exports = class User extends CocoModel
   skipHeroSelectOnStudentSignUp: -> features?.classroomItems ? false
   useDexecure: -> not (features?.chinaInfra ? false)
   useSocialSignOn: -> not (features?.chinaUx ? false)
+  isTarena: -> features?.Tarena ? false
```
Comment (shubhi1092) [inline app/models/User.coffee]: Do we need to add it on server also?

Comment (AndrewJakubowicz) [reply app/models/User.coffee]: We just need to add a new feature flag under the `user` type with the name `Tarena`.
The server has a middleware which will be able to correctly send `features.Tarena` for clients created by the Tarena API client. Just need to set it to true for that API client and it'll affect all the users.

Comment (AndrewJakubowicz) [reply app/models/User.coffee]: As far as I know this will be a direct mongo database addition as I don't think we've created a ui to add and remove feature flags.

File: app/templates/base-flat.jade (hunk 2 above)
Comment (AndrewJakubowicz) [reply app/templates/base-flat.jade]: Thanks for catching this bug!

File: app/models/User.coffee (hunk 4 above)
Comment (shubhi1092) [reply app/models/User.coffee]: Yeah that sounds right, thanks for clarifying. We may also need to add it to the features collection in database.
